src/node_file.cc

```diff
@@ -566,17 +569,21 @@ static void InternalModuleReadFile(const FunctionCallbackInfo<Value>& args) {
   CHECK_EQ(0, uv_fs_close(loop, &close_req, fd, nullptr));
   uv_fs_req_cleanup(&close_req);
 
-  size_t start = 0;
-  if (chars.size() >= 3 && 0 == memcmp(&chars[0], "\xEF\xBB\xBF", 3)) {
-    start = 3;  // Skip UTF-8 BOM.
-  }
+  if (numchars < 0) {
+    args.GetReturnValue().Set(Undefined(env->isolate()));
```

Comment:
Rather than returning `undefined`.... can we throw here? Yes, I know that's semver-major but it's better than an abort and it's likely more correct than `undefined`.